src/locomoset/config/config_classes.py

```diff
@@ -0,0 +1,254 @@
+
+
+class Config(ABC):
+
+    """Base class for config objects
+
+    Attributes:
+        model_name: Name of the HuggingFace model to fine-tune.
+        dataset_name: Name of the HuggingFace dataset to use for fine-tuning.
+        run_name: Name of the run (used for wandb/local save location), defaults to
+            {dataset_name}_{model_name}.
+        random_state: Random state to use for train/test split and training.
+        dataset_args: Dict defining "train_split" and "val_split" (optional), defaults
+            to {"train_split": "train"}.
+        training_args: Dict of arguments to pass to TrainingArguments.
+        use_wandb: Whether to use wandb for logging.
+        wandb_args: Arguments to pass to wandb.init.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        dataset_name: str,
+        random_state: int | None = None,
+        config_gen_dtime: str | None = None,
+        caches: dict | None = None,
+        wandb_args: dict | None = None,
+        use_wandb: bool = False,
+        run_name: str | None = None,
+    ) -> None:
+        super().__init__()
+        self.model_name = model_name
+        self.dataset_name = dataset_name
+        self.random_state = random_state
+        self.config_gen_dtime = config_gen_dtime
+        self.caches = caches
+        self.use_wandb = use_wandb
+        self.wandb_args = wandb_args or {}
+        self.run_name = run_name or f"{dataset_name}_{model_name}".replace("/", "-")
+
+    def init_wandb(self) -> None:
+        """Initialise a wandb run if the config specifies to use wandb and a run has not
+        already been initialised.
+
+        If name, group and job_type and not specificied in the input config then they
+        are set as:
+                name: run_name
+                group: data_set_name_config_gen_dtime OR data_set_name
+                job_type: misc
+        """
+        if not self.use_wandb:
+            warnings.warn("Ignored wandb initialisation as use_wandb=False")
+            return
+        if wandb.run is not None:
+            raise ValueError("A wandb run has already been initialised")
+
+        wandb.login()
+        wandb_config = copy(self.wandb_args)
+
+        if "log_model" in wandb_config:
+            # log_model can only be specified as an env variable, so we set the env
+            # variable then remove it from the init args.
+            os.environ["WANDB_LOG_MODEL"] = wandb_config["log_model"]
+            wandb_config.pop("log_model")
+
+        # set default names for any that haven't been specified
+        if "name" not in wandb_config:
+            wandb_config["name"] = self.run_name
+        if "group" not in wandb_config:
+            if self.config_gen_dtime is not None:
+                wandb_config["group"] = f"{self.dataset_name}_{self.config_gen_dtime}"
+            else:
+                wandb_config["group"] = f"{self.dataset_name}"
+        if "job_type" not in wandb_config:
```

Comment:
I had left it as `misc` in case we ran something other than the metrics or training experiment, but probably makes sense to raise an error instead, I've set it to that now.